File: 1324. Print Words Vertically.py
class Solution:
    def printVertically(self, s: str) -> List[str]:
        ans, word, pre = [], s.split(' '), 1
        
        while pre:
            pre, ins = 0, ''
            
            for i in range(len(word)-1, -1, -1):
                if not (pre or word[i]):
                    continue
                
                if word[i]:
                    ins, word[i] = word[i][0]+ins, word[i][1:]
                
                else:
                    ins = ' '+ins
                    
                pre = 1
            
            ans.append(ins)
        
        ans.pop()
        
        return ans

File: 1324. Print Words Vertically.js
/**
 * @param {string} s
 * @return {string[]}
 */
var printVertically = function(s) {
    let [ans, word, pre] = [[], s.split(' '), 1];
    
    while(pre){
        let ins = '';
        pre = 0

        for(let i=word.length-1;i>-1;i--){
            if(!(pre || word[i])){
                continue;
            }else if(word[i]){
                [ins, word[i]] = [word[i][0]+ins, word[i].slice(1)];
            }else{
                ins = ' '+ins;
            }
            pre = 1;
        }
        ans.push(ins);
    }
    ans.pop();

    return ans;
};